Added Platform IO, Added SSL, Added an RGB LED

SittingDuc opened the pull request:
Hello, you do not know me from a random duc on the street, but I have forked your heatpump project and adapted it to my own controller, which has an RGB LED (red for room warm, blue for room cold, green for just right and yellow for sleeping) and uses MQTTS encryption.

Take two: I have put the files back in mitsucon/, because PlatformIO is smarter than I am. The diff is still ugly, it might need white-space suppression to show the actual changes instead of flagging every single line.

In the spirit of Open Source, I offer my changes back upstream. They are here if you want them.

Thanks!

SittingDuc committed:
Migrating code Milestone

Mostly complete copy of code edits from https://github.com/SittingDuc/HeatPump into this here mitsucon project

Not yet tested in hardware, but VSCode compiles cleanly for me, so into the tree it goes

SittingDuc committed:
1. Migrated from AxTLS to BearSSL, which changed MQTTS authentication a bit
2. added support for NTP, because X.509 validation appreciates that
3. Commented every #else and #endif for more readability
4. Commented the .h more heavily
5. Added debug, removed debug, fixed bugs, field-tested in Hardware

This now resembles the code running on my little PCBs

SittingDuc committed:
Fix #ifdef preventing compile when #undef HAVE_MQTTS

SittingDuc committed:
Put code back in mitsucon/ directory, because platformio can be told where to look for sources
Reduce upstream churn by not making arbitrary changes

unixko commented:
Hello @SittingDuc. Thanks for contribution. I did backlog clean up task today but I also moved files location. I need to learn about PlatformIO first so in this mean time I will put your fork link in README along with other Mitsubishi controller projects you may be interested. If possible please share a picture of your final installation with heat pump and LED together. It is very interesting!

SittingDuc committed:
Update README, add pictures, describe fork

SittingDuc committed:
whitespace fixes

SittingDuc commented:
Hello, I have updated my fork of README.md with pictures of the adapter PCB and of a heatpump with the LED in place. Once installed, it isn't actually very charismatic, just a glowing light..

SittingDuc committed:
ArduinoOTA fixes. Upstream fixes.

(manually) Pulling in changes from upstream.
Tweaked BearSSL verify to eliminate "deprecated" message
Tweaked ArduinoOTA code to be more likely to work
